core/services/vrf/v2/listener_v2.go

```diff
@@ -1557,28 +1587,226 @@ func (lsn *listenerV2) runRequestHandler(pollPeriod time.Duration, wg *sync.Wait
 	}
 }
 
-func (lsn *listenerV2) runLogListener(unsubscribes []func(), minConfs uint32, wg *sync.WaitGroup) {
+func (lsn *listenerV2) runLogListener(
+	pollPeriod time.Duration,
+	minConfs uint32,
+	wg *sync.WaitGroup,
+	lastProcessedBlock int64,
+) {
 	defer wg.Done()
-	lsn.l.Infow("Listening for run requests",
+	lsn.l.Infow("Listening for run requests via log poller",
 		"minConfs", minConfs)
+	ticker := time.NewTicker(utils.WithJitter(pollPeriod / 2))
+	defer ticker.Stop()
 	for {
 		select {
 		case <-lsn.chStop:
-			for _, f := range unsubscribes {
-				f()
-			}
 			return
-		case <-lsn.reqLogs.Notify():
-			// Process all the logs in the queue if one is added
-			for {
-				lb, exists := lsn.reqLogs.Retrieve()
-				if !exists {
-					break
-				}
-				lsn.handleLog(lb, minConfs)
+		case <-ticker.C:
+			err := func() error {
+				ctx, cancel := lsn.chStop.NewCtx()
+				defer cancel()
+				return lsn.pollLogs(ctx, minConfs, lastProcessedBlock)
+			}()
+			// might be a retryable error
+			if err != nil {
+				lsn.l.Errorw("error polling vrf logs", "err", err)
+				continue
+			}
+			lastProcessedBlock, err = lsn.updateLastProcessedBlock()
+			if err != nil {
+				lsn.l.Errorw("error updating last processed block", "err", err)
+			}
+		}
+	}
+}
+
+func (lsn *listenerV2) initializeLastProcessedBlock(latestBlock logpoller.LogPollerBlock) (lastProcessedBlock int64, err error) {
+	lp := lsn.chain.LogPoller()
+	fromTimestamp := time.Now().UTC().Add(-24 * time.Hour)
+	// get randomness requested logs with the appropriate keyhash
+	// keyhash is specified in topic1
+	requests, err := lp.IndexedLogsCreatedAfter(
+		lsn.coordinator.RandomWordsRequestedTopic(), // event sig
+		lsn.coordinator.Address(),                   // address
+		1,                                           // topic index
+		[]common.Hash{lsn.job.VRFSpec.PublicKey.MustHash()}, // topic values
+		fromTimestamp, // from time
+		1,             // confs
```

Comment:
I think in both reqs and fulfillments you only want finalized logs here i.e. logpoller.Finalized, o/w you'd may mark lastProcessed on a block that gets reorged

Reply:
Oh I see, then I think we want the logic where if earliest unprocessed block > latest finalized block, we still have latest finalized block as the last processed right?

Reply:
Changed this to use logpoller.Finalized, thanks for the catch. Also, in updateLastProcessedBlock, I check if the earliest unprocessed request block is > than latest finalized block, and if so I use latest finalized as last processed, since the earliest unprocessed request block is not finalized.